rules/TypeDeclaration/Rector/ClassMethod/ReturnTypeFromStrictNewArrayRector.php

```diff
@@ -147,7 +147,7 @@ private function processAddArrayReturnType(
         ClassMethod|Function_|Closure $node,
         Type $returnType
     ): ClassMethod|Function_|Closure|null {
-        if (! $returnType->isArray()->yes()) {
+        if (! $returnType instanceof ArrayType && ! $returnType instanceof ConstantArrayType) {
```

Comment:
Won't this stop working on a union of arrays?

Reply:
Or on non-empty-array, which is a intersection type

Reply:
I talked with @TomasVotruba about this, and more and more we used PHPStan method from Type, it just cause more bugs, so we decide to use instanceof if the variable of call is used next, that's allow for improvement for more types later, rather than cause crash because of method not exists in target PHPStan type.

Reply:
Yeah, as soon as you use methods which are not part of the Type interface you hurt PHPStan guarantees.

It would be great if we could get rid of calls to methods which only exist on certain types

Reply:
see https://github.com/rectorphp/rector-src/pull/6478